superset/alfred/utils.py

```diff
@@ -0,0 +1,350 @@
+                    if "Event Organization" in log
+                    else "Unknown Department"
+                )
+                if dept not in dept_map:
+                    dept_map[dept] = event_count
+                    event_count += 1
+
+                # Set the description per event.
+                # If logs supply 'Event Info' it will be grabbed from all the logs, uniqued and sorted into the description
+                if dept not in info_map:
+                    info_map[dept] = "\n".join(
+                        sorted(
+                            list(
+                                set(
+                                    [
+                                        d["Event Info"]
+                                        for d in data_set
+                                        if "Event Info" in d
+                                        and (
+                                            (
+                                                dept != "Unknown Department"
+                                                and "Event Organization" in d
+                                                and dept == d["Event Organization"]
+                                            )
+                                            or (
+                                                dept == "Unknown Department"
+                                                and "Event Organization" not in d
+                                            )
+                                        )
+                                    ]
+                                )
+                            )
+                        )
+                    )
+                    log["event description"] = (
+                        "{" + markup + "}" + f"{dept}\n{info_map[dept]}".strip()
+                    )
+                elif dept in info_map:
+                    log["event description"] = (
+                        "{" + markup + "}" + f"{dept}\n{info_map[dept]}".strip()
+                    )
+                else:
+                    log["event description"] = "{" + markup + "}" + f"{dept}".strip()
+
+                # Set the Event ID
+                log["new event id"] = f"LS-<{event_type}_{dept_map[dept]}>"
+
+        else:
+            if "new event id" not in log:
+                log["new event id"] = f"LS-<{event_type}_1>"
+
+    # Create Retention and Stage it
+    logger.info(f"Creating Retention...")
+    retention = RetentionInstance(
+        alfred,
+        f"Fission Retention - {datetime.now()}",
+        data_set,
+        metadata_system,
+        server_side_mapping_name="hogwarts.harmonized.eml_metadata",
+    )
+    ready = retention.stage()
+
+    # logger.info the result
+    logger.info(f"Retention Ready: {ready}.")
+    logger.info(f"Retention Messages: {retention.retention.message_summary}.")
+
+    # Return the URL
+    retention_url = (
+        InstanceUtil.build_url(alfred_instance)
+        .replace(":9488", "")
```

Comment:
Just a little suspicious about this. Why is that hardcoded port kicking around?

Reply:
I guess thats from the alfred library, it returns a link that includes the port number in it. I am not sure if/why we need to remove it

Reply:
nevermind its just a trick to reuse their build url function, but for the UI link as well. Leaving it as is